src/tmpfiles/tmpfiles.c

```diff
@@ -848,6 +848,42 @@ static int fd_set_perms(Item *i, int fd, const char *path, const struct stat *st
         return label_fix(path, 0);
 }
 
+static int log_unsafe_transition(const char *path) {
+        char *a, *b;
+        int r;
+
+        assert(path);
+
+        a = strdupa(path);
+
+        for (b = a ;; *b = '/') {
+
+                b = strchr(b + 1, '/');
+                assert(b);
+                *b = '\0';
+
+                r = chase_symlinks(a, NULL, CHASE_SAFE, NULL);
+                if (r < 0) {
+                        assert(r == -EPERM);
+                        break;
+                }
+        }
+
+        /* Put 'b' on the last component of the last checked path. */
+        while (*b != '/')
+                b--;
+        *b++ = '\0';
+
+        /* Move 'a' on the path component right before 'b' or set it to NULL
+         * if 'a' is the root directory. */
+        a = strrchr(a, '/');
+        if (a)
+                a++; /* skip '/' */
+
+        return log_error_errno(r, "Unsafe path transition %s → %s in %s, refusing.",
+                               empty_to_root(a), b, path);
```

Comment:
this is not right. Let's say you have a line that wants to configure "/foo", and that is a symlink pointing to /bar/quux. And the unsafe transition is somewhere when resolving /var/quux: your code would not catch that and would be quite misldeading then.

If you want to do something like this I think adding a new flag CHASE_WARN or so to chase_symlinks() would be the better approach, and generate the warning right from where the full understanding of the concept is.